### DISCUSSION TITLE
Please Add Datasource for The Buf Schema Registry

### DISCUSSION THREAD

**Discussion by isso-719** [renovatebot/renovate#33268] [Suggest an Idea]
### Tell us more.

I am developing with gRPC.
I want Renovate to automatically update the Protobuf Module that exists in The Buf Schema Registry.

For example, to get the version of `grpc-ecosystem/gateway`, I have to send the following request.

```bash
curl "https://buf.build/buf.alpha.registry.v1alpha1.ResourceService/GetResourceByName" \
-H "Content-Type: application/json" \
-d '{
  "owner": "grpc-ecosystem",
  "name": "gateway"
}'
```

However, Custom Datasource cannot include request bodies.
Thus, I would like to see a Datasource for The Buf Schema Registry implemented.

**Comment by rarkins** [comment]
@secustor wdyt about allowing request bodies for custom datasources?

**Reply by secustor** [reply]
It is fine for me, but we have some things to consider so Renovate is not used to DDOS random APIs. 

The easiest way would be to limit body size l.

**Comment by rarkins** [comment]
Added feature request for custom request bodies: https://github.com/renovatebot/renovate/issues/33473

**Comment by anubhuti2023** [comment]
Is this feature created ?

**Reply by anubhuti2023** [reply]
https://github.com/renovatebot/renovate/discussions/36562 is this discussion related to this feature addition?
